uaclient/entitlements/repo.py

```diff
@@ -68,6 +67,7 @@ def enable(self, *, silent_if_inapplicable: bool = False) -> bool:
         if not self.can_enable(silent=silent_if_inapplicable):
             return False
         if not self.setup_apt_config():
+            self._cleanup()  # Cleanup any apt sources/auth artifacts
```

Comment:
I wonder if it might be easier to implement this for FIPS if, instead of calling the broad `_cleanup` function we instead modified `setup_apt_config` to do its own cleanup on failure before it returns False. So it's effectively transactional (i.e. it will either modify the system and return True or it will fail to modify the system, undo any changes it made and return False).

In retrospect, I think introducing `_cleanup` may have been a mistake on my part, because if we call `_cleanup` in all places where we need to clean up, I worry that it has to be written in a way that can dynamically detect and correctly handle every possible intermediate state that we want to clean up. (That's a needlessly hard problem to solve if we can avoid it.)

Reply:
Instead of using _cleanup here, we can use remove_apt_config() which fips hasn't overridden. If remove_apt_config isn't idempotent, I believe we should make it idempotent to handle intermittent install conditions.